src/extraction/FStar.Extraction.ML.Modul.fs

```diff
@@ -983,25 +983,83 @@ let rec extract_sig (g:env_t) (se:sigelt) : env_t * list<mlmodule1> =
+                     (fun () -> Env.postprocess env tau lb.lbtyp lb.lbdef)
+                     (Some (Ident.string_of_lid (Env.current_module env)))
+                     "FStar.Extraction.ML.Module.post_process_for_extraction"
+              in
               { lb with lbdef = lbdef }
+            in
+            match post_tau with
+            | None -> lbs
+            | Some tau -> fst lbs, List.map (postprocess_lb tau) (snd lbs)
           in
-          let lbs = (fst lbs,
-                      (match post_tau with
-                       | Some tau -> List.map (postprocess_lb tau) (snd lbs)
-                       | None -> (snd lbs)))
+          let maybe_normalize_for_extraction lbs = 
+            let norm_steps =
+              match U.extract_attr' PC.normalize_for_extraction_lid attrs with
+              | None -> None
+              | Some (_, (steps, None)::_) ->
+                let steps = 
+                  //just normalizing the steps themselves, so that the user
+                  //does not have to write a literal at every use of the attribute
+                  N.normalize 
+                    [Env.UnfoldUntil delta_constant; Env.Zeta; Env.Iota; Env.Primops]
+                    (tcenv_of_uenv g)
+                    steps
+                in
+                begin
+                match Cfg.try_unembed_simple (EMB.e_list EMB.e_norm_step) steps with
+                | Some steps -> 
+                  Some (Cfg.translate_norm_steps steps)
+                | _ -> 
+                  Errors.log_issue 
+                      se.sigrng
+                      (Errors.Warning_UnrecognizedAttribute,
+                       BU.format1
+                         "Ill-formed application of 'normalize_for_extraction': normalization steps '%s' could not be interpreted"
+                         (Print.term_to_string steps));
+                  None
+                end
+              | Some _ ->
+                  Errors.log_issue 
+                      se.sigrng
+                      (Errors.Warning_UnrecognizedAttribute,
+                       "Ill-formed application of 'normalize_for_extraction'");
+                  None
+            in
+            let norm_one_lb steps lb =
+              let env = tcenv_of_uenv g in
+              let env = {env with erase_erasable_args=true} in
+              let lbd = 
+                Profiling.profile
+                     (fun () -> N.normalize steps env lb.lbdef)
+                     (Some (Ident.string_of_lid (Env.current_module env)))
+                     "FStar.Extraction.ML.Module.normalize_for_extraction"
+              in
+              { lb with lbdef = lbd }
+            in
+            match norm_steps with
+            | None -> lbs
+            | Some steps ->
+              fst lbs, List.map (norm_one_lb steps) (snd lbs)
           in
           let ml_let, _, _ =
+            let lbs = maybe_normalize_for_extraction (maybe_postprocess_lbs lbs) in
```

Comment:
@aseemr @tahina-pro 

This is the bit that runs postprocess_for_extraction first, then normalize_for_extraction. Later in the pipeline, we'll do another norm call with inline for extraction.